src/broker/cas_cgw.c

```diff
@@ -1770,6 +1770,12 @@ cgw_get_num_cols (SQLHSTMT hstmt, SQLSMALLINT * num_cols)
 
   SQL_CHK_ERR (hstmt, SQL_HANDLE_STMT, err_code = SQLNumResultCols (hstmt, num_cols));
 
+  if (*num_cols == 0)
+    {
+      SQL_CHK_ERR (hstmt, SQL_HANDLE_STMT, err_code = SQLExecute (hstmt));
+      SQL_CHK_ERR (hstmt, SQL_HANDLE_STMT, err_code = SQLNumResultCols (hstmt, num_cols));
+    }
+
   return (int) err_code;
```

Comment:
Maybe you combined two different issues in one jira issue and a pull request. Next time, obey the rule that a jira issue should only cover one issue at a time.

Reply:
I will definitely separate them from next time.